web/src/filter.ts

```diff
@@ -297,6 +297,7 @@ export class Filter {
     _can_mark_messages_read?: boolean;
     requires_adjustment_for_moved_with_target?: boolean;
     narrow_requires_hash_change: boolean;
+    cached_sorted_terms_for_comparison?: string[] = undefined;
```

Comment:
Does this require some interaction with `requires_adjustment_for_moved_with_target`? I can't recall if that ends up making a new `Filter` operator or uses the same one, where we'd need to clear this cache.

Reply:
Yes, reset in `setup_filter`. Nice catch!